receiver/elasticsearchreceiver/scraper.go

```diff
@@ -95,12 +97,28 @@ func (r *elasticsearchScraper) scrape(ctx context.Context) (pmetric.Metrics, err
 
 	now := pcommon.NewTimestampFromTime(time.Now())
 
+	r.getVersion(ctx)
 	r.scrapeNodeMetrics(ctx, now, errs)
 	r.scrapeClusterMetrics(ctx, now, errs)
 
 	return r.mb.Emit(), errs.Combine()
 }
 
+// scrapeVersion gets and assigns the elasticsearch version number
+func (r *elasticsearchScraper) getVersion(ctx context.Context) {
+	versionResponse, err := r.client.Version(ctx)
+	if err != nil {
+		return
+	}
+
+	esVersion, err := version.NewVersion(versionResponse.Version.Number)
+	if err != nil {
+		return
+	}
```

Comment:
If either of these fail, does `r.version` remain nil? This would panic when checking `if r.version.GreaterThanOrEqual(es10) {`

Reply:
I believe It would never get to the `if r.version.GreaterThanOrEqual` check.
If r.version were to fail then that means the default endpoint, for example a GET request on `http://localhost:9200` would not work. That would also mean `http://localhost:9200/_nodes/_all/stats/{nodeStatsMetrics}/{nodeStatsIndexMetrics}` would not work as well and [return](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/elasticsearchreceiver/scraper.go#L110-L113) early. 

I can add a [Check](https://pkg.go.dev/github.com/hashicorp/go-version#Constraint.Check) version for more assurance.

Reply:
If an error is received then perhaps the scraper should end early with the returned error, most likely a `Malformed version` error and never call `r.scrapeNodeMetrics` and `r.scrapeClusterMetrics` then actually.

Reply:
Updated with second mentioned approach